packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -60,13 +61,17 @@ function hydrateColumns(
   columnTypes: GridColumnTypesRecord,
   withCheckboxSelection: boolean,
   logger: Logger,
+  getLocaleText: <T extends GridTranslationKeys>(key: T) => GridLocaleText[T],
 ): GridColumns {
   logger.debug('Hydrating GridColumns with default definitions');
   const mergedColTypes = mergeGridColTypes(getGridDefaultColumnTypes(), columnTypes);
   const extendedColumns = columns.map((c) => ({ ...getGridColDef(mergedColTypes, c.type), ...c }));
 
   if (withCheckboxSelection) {
-    return [gridCheckboxSelectionColDef, ...extendedColumns];
+    const checkboxSelection = { ...gridCheckboxSelectionColDef };
+    checkboxSelection.headerName = getLocaleText('checkboxSelectionHeaderName');
+    checkboxSelection.description = getLocaleText('checkboxSelectionDescription');
```

Comment:
`description` is defined as "The description of the column rendered as tooltip if the column header name is not fully displayed". It doesn't seem to be relevant.

How about we simplify it to?

```suggestion
```